cpp/src/column/column_device_view.cu

```diff
@@ -32,22 +32,82 @@ column_device_view::column_device_view(column_view source)
 
 // Free device memory allocated for children
 void column_device_view::destroy() {
-  // TODO Implement once support for children is added
+  // TODO Needs to handle grand-children
+  if( d_children )
+    RMM_FREE(d_children,0);
+  delete this;
 }
 
-// Construct a unique_ptr that invokes `destroy()` as it's deleter
-auto column_device_view::create(column_view source, cudaStream_t stream) {
-  size_type num_descendants{count_descendants(source)};
-  if (num_descendants > 0) {
-    CUDF_FAIL("Columns with children are not currently supported.");
+// For use with inplace-new to pre-fill memory to be copied to device
+column_device_view::column_device_view( column_view source, ptrdiff_t h_ptr, ptrdiff_t d_ptr )
+    : detail::column_device_view_base{source.type(),       source.size(),
+                                      source.head(),       source.null_mask(),
+                                      source.null_count(), source.offset()},
+      _num_children{source.num_children()}
+{
+  if( _num_children > 0 )
+  {
+    column_device_view* h_column = reinterpret_cast<column_device_view*>(h_ptr);
+    column_device_view* d_column = reinterpret_cast<column_device_view*>(d_ptr);
+    d_children = d_column;
+    for( size_type idx=0; idx < _num_children; ++idx )
+    { // inplace-new each child
+      column_view child = source.child(idx);
+      CUDF_EXPECTS( child.num_children()==0, "column grand-children not currently supported");
+      new(h_column) column_device_view(child);
+      h_column++;
+      //d_column++;
+    }
   }
+}
 
-  auto deleter = [](column_device_view* v) { v->destroy(); };
+// For use with inplace-new to pre-fill memory to be copied to device
+mutable_column_device_view::mutable_column_device_view( mutable_column_view source, ptrdiff_t h_ptr, ptrdiff_t d_ptr )
+    : detail::column_device_view_base{source.type(),       source.size(),
+                                      source.head(),       source.null_mask(),
+                                      source.null_count(), source.offset()}
+{}
 
+// Construct a unique_ptr that invokes `destroy()` as it's deleter
+std::unique_ptr<column_device_view, std::function<void(column_device_view*)>> column_device_view::create(column_view source, cudaStream_t stream) {
+  //size_type num_descendants{count_descendants(source)};
```

Comment:
This is a change from 2207 code. It is difficult to see the changes to files introduced or modiffied by 2207 for this PR. So I left this code commented out for review for now. It will be easier to find and review these once 2207 is merged.